src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs

```diff
@@ -0,0 +1,269 @@
+                await CanReadAsync(context, requirement, resources, org);
+                break;
+
+            case not null when requirement == BulkCollectionOperations.ReadAccess:
+                await CanReadAccessAsync(context, requirement, resources, org);
+                break;
+
+            case not null when requirement == BulkCollectionOperations.Update:
+            case not null when requirement == BulkCollectionOperations.ModifyAccess:
+                await CanManageCollectionAccessAsync(context, requirement, resources, org);
+                break;
+
+            case not null when requirement == BulkCollectionOperations.Delete:
+                await CanDeleteAsync(context, requirement, resources, org);
+                break;
+        }
+    }
+
+    private async Task CanCreateAsync(AuthorizationHandlerContext context, IAuthorizationRequirement requirement,
+        CurrentContextOrganization? org)
+    {
+        // If the limit collection management setting is disabled, allow any user to create collections
+        // Otherwise, Owners, Admins, and users with CreateNewCollections permission can always create collections
+        if (org is
+        { LimitCollectionCreationDeletion: false } or
+        { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } or
+        { Permissions.CreateNewCollections: true })
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        // Allow provider users to create collections if they are a provider for the target organization
+        if (await _currentContext.ProviderUserForOrgAsync(_targetOrganizationId))
+        {
+            context.Succeed(requirement);
+        }
+    }
+
+    private async Task CanReadAsync(AuthorizationHandlerContext context, IAuthorizationRequirement requirement,
+        ICollection<Collection> resources, CurrentContextOrganization? org)
+    {
+        // Owners, Admins, and users with EditAnyCollection or DeleteAnyCollection permission can always read a collection
+        if (org is
+        { LimitCollectionCreationDeletion: false } or
+        { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } or
+        { Permissions.EditAnyCollection: true } or
+        { Permissions.DeleteAnyCollection: true } or
+        { Permissions.CreateNewCollections: true } or
+        { Permissions.ManageUsers: true })
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        // The acting user is a member of the target organization,
+        // ensure they have access for the collection being read
+        if (org is not null)
+        {
+            var canManageCollections = await HasCollectionAccessAsync(resources, org, requireManagePermission: false);
+            if (canManageCollections)
+            {
+                context.Succeed(requirement);
+                return;
+            }
```

Comment:
Variable name `canManageCollections` does not match argument `requireManagePermissions: false`. Which should it be?

Comparing to the current logic in both `Get` and `GetDetails`, I think this should be `requireManagePermission: true` for both. Both seem to require manager permissions (they generally trickle down to `currentContext.ViewAssignedCollections` -> `currentContext.EditAssignedCollections` which is a manager check). But please double check me.

Reply:
You're right, I've also reviewed these and removed the `requireManagePermission` parameter as it was no longer used.

Reply:
I double checked and I think you're right. I got confused (you can understand why 😵‍💫)